Fixes per review notes

kateyurasova opened the pull request.

RapeikaD committed:
UpdatesV0.3

+Points 1-3 from list

RapeikaD committed:
UpdateV0.3.2

Latest fixes

RapeikaD committed:
v0.7

Rework of primary automated tests.

kateyurasova commented on node_modules/.bin/allure:
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
+
```
Remove note_modules folder from git version control. You may do it by adding gitignore file. More details about it here: https://www.atlassian.com/git/tutorials/saving-changes/gitignore

RapeikaD committed:
v0.7.2

+Page Object
+gitignore
+fixed launch issues
+small improvement for readme
+update dependencies for package.json
+commented code removed
+added comments about what was happened in the program
+atempt to remove node_modules from github

katsiaryna-yurasova-zefr commented on loginPage.js:
```diff
@@ -0,0 +1,25 @@
+let loginPage = function () {
+
```
Сгруппируй файлы в папки tests, pages, data

katsiaryna-yurasova-zefr commented on ATT01444.gitignore:
```diff
@@ -0,0 +1,5 @@
+screenshots/
```
Имя файла выглядит странно, не опознается гитом. Должно быть .gitignore.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
-
+	beforeEach(function() {
 		browser.waitForAngularEnabled(false);
+		browser.get(data.site);//add in page Object
```
Код по сути является открытием страницы loginPage:
                browser.waitForAngularEnabled(false);
		browser.get(data.site);//add in page Object
Вынеси в функцию open(), вызывай как loginPage.open()

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+		loginPage.clickNext2();
 		browser.wait(element(by.css(".T-I")).isPresent())
-  		browser.sleep(5000)
+  		browser.sleep(data.timer)
```
Код от строки 15 до текущий по сути - процесс логина. Вынеси в функцию login c параметрами login и password.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+	it("should send email and verify Sent email", function() {
+		//заходим в почту, создаем письмо и отправляем 2му пользователю
+		loginPage.enterName(data.EmailOne);
+		loginPage.clickNext1();
```
clickNext1 (а также clickNext2) не самое удачное название для фукнции. Переименуй во что-то более подходящее.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
-		element(by.id('identifierNext')).click();
+	it("should send email and verify Sent email", function() {
+		//заходим в почту, создаем письмо и отправляем 2му пользователю
+		loginPage.enterName(data.EmailOne);
```
EmailOne не самое лучшее название для поля. По смыслу теста это имэйл отправителя. Переименуй например в senderEmail. Для получателя используй по аналогии receiverEmail.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
 		browser.wait(element(by.css(".T-I")).isPresent())
-  		browser.sleep(5000)
+  		browser.sleep(data.timer)
   		browser.get("https://mail.google.com/mail/#inbox?compose=new")
```
Вынеси в функцию страницы inbox.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+		inbox.TypeTitle(RandomTitle)
+		inbox.TypeDescription(data.BodyText)
 		browser.actions().keyDown(protractor.Key.CONTROL).sendKeys('\uE007').perform();
 		browser.get("https://mail.google.com/mail/#sent")
```
вынеси в функцию страницы inbox

katsiaryna-yurasova-zefr replied:
Избегай дублирования кода. https://mail.google.com/mail используется в тесте несколько раз. Выноси в переменную класса страницы InboxPage

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+
+		data = require("./data.js");		
+		loginPage = require("./loginPage.js");
+		inbox = require("./inbox.js");
```
Придерживайся единого паттерна для именования станиц. Переименуй inbox в InboxPage.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+	beforeEach(function() {
 		browser.waitForAngularEnabled(false);
+		browser.get(data.site);//add in page Object
+	});
```
Нужно убрать ; по всему коду.
В ES6 они не нужны.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
 		for(var i = 0; i < headerList.length; i ++) {
 			if (headerList[i] == RandomTitle) {
 				expect(headerList.get(i).getText()).toBe(RandomTitle)
-				expect(list.get(i).getText()).toBe(' - You are the best!')
```
Вынести код по проверке title и body в функции класса страницы.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+		
+		//генерируем уникальное имя для письма и вводим шаблонный текст письма
+		var number = Math.round((Math.random() * 100))
 		var RandomTitle = "Hello dear client # " + number
```
Вынести строку в тестовые данные.

katsiaryna-yurasova-zefr commented on spec.js:
```diff
@@ -1,68 +1,82 @@
+		browser.sleep(data.timer)
+		
+		//генерируем уникальное имя для письма и вводим шаблонный текст письма
+		var number = Math.round((Math.random() * 100))
```
вместо var используй let

RapeikaD committed:
v0.7.3

+proposed fixes
To do:
Readme

katsiaryna-yurasova-zefr commented on .gitignore.gitignore:
```diff
@@ -0,0 +1,6 @@
+screenshots/
```
Имя файла должно быть .gitignore а не .gitignore.gitignore

katsiaryna-yurasova-zefr commented on Pages/inbox.js:
```diff
@@ -0,0 +1,35 @@
+
+	this.newDraft = function (delay) {
+        browser.get("https://mail.google.com/mail/#inbox?compose=new");
+        browser.sleep(delay);
```
Замени delay на ожидание появление чего-то что будет являться признаком того что страница загрузилась. 
Например элемент с каким-либо текстом или просто какой-либо элемент присущий только этой странице.
Пример ищи в проекте https://github.com/MariyaMitko/FoodMe
например
browser.wait(EC.textToBePresentInElement(controlMessage, `${numberOfResults} restaurants found`))

katsiaryna-yurasova-zefr commented on Tests/spec.js:
```diff
@@ -0,0 +1,80 @@
+    loginPage.enterPassword(password);
+    loginPage.submitPassword();
+    browser.wait(element(by.css(".T-I")).isPresent());
+    browser.sleep(data.timer);
```
Убрать sleep. Если без него не работает, заменить на ожидание того, что является признаком того что логин прошел успешно.

RapeikaD committed:
Last Update

+fixes

katsiaryna-yurasova-zefr commented on .idea/automation.iml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
```
Дима, ты конечно переименовал gitignore но нужно было удалить закомитанные файлы прежде. Иначе вот как ты видишь они тут. Нужно было действовать как тогда: 
![image](https://user-images.githubusercontent.com/41969640/48258218-a2242200-e425-11e8-82fc-80240cdfcbee.png)

katsiaryna-yurasova-zefr commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+screenshots/
+node_modules/
+allure-results/
```
У тебя есть папочка allure-results. Ура. Это значит, что репортер работает. 
Но есть у тебя наверняка вопрос, что делать с этими файликами вообще.
Чтобы на основе этих xml сгенерировать отчет, нужно поставить пакет allure-commandline:
https://www.npmjs.com/package/allure-commandline
Как с ним работать - я покажу. Звони как поставишь и будешь готов.

RapeikaD committed:
v0.7.4

 Excess folders are removed

RapeikaD committed:
v0.8

-FIxes for Readme.
-All "sleep"  was removed. Timer variable removed as well.
-Login fucntion was improved. Excess code was deleted.
-Locators update.
-Allure command line connected.

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+
+# Please first install:
+1. Install Protractor
+2. Install Node.js
```
Add information about version of Node JS

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+You can find Required Information below.
+
+# Please first install:
+1. Install Protractor
```
Add links to pages with dowloads (optional), specify version.

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+# Please first install:
+1. Install Protractor
+2. Install Node.js
+3. Install Java
```
versoin

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+2. Install Node.js
+3. Install Java
+4. Install Allure (npm i jasmine-allure-reporter)
+    4.1 Install Allure commandline (npm install -g allure-commandline --save-dev)
```
Ненужная инструкция. Установка аллюр включено в npm install, происходит на основании прописанного в package.json

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+3. Install Java
+4. Install Allure (npm i jasmine-allure-reporter)
+    4.1 Install Allure commandline (npm install -g allure-commandline --save-dev)
+5. Install Chance library (npm install chance)
```
ненужная инструкция (package.json)

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+To add components type: npm install protractor --save-dev
+To update webdriver manager type: webdriver-manager update
+
+npm update --save-dev
```
Используй блоки
```
npm update --save-dev
```

kateyurasova commented on README.md:
```diff
@@ -1,2 +1,37 @@
+
+npm update --save-dev
+
+
```
Добавить инструкции по генерации отчета

kateyurasova commented on Pages/inboxPage.js:
```diff
@@ -0,0 +1,68 @@
+    };
+
+    this.newDraft = function () {
+        browser.get("https://mail.google.com/mail/#inbox?compose=new");
```
Значение урла https://mail.google.com/mail используется много раз в проекте.
Нужно вынести в конфиг и читать это значение оттуда.

kateyurasova commented on Pages/inboxPage.js:
```diff
@@ -0,0 +1,68 @@
+    };
+    //looks like this functions are not working
+    this.createdEmailCheck = function (RandomTitle, BodyText) {
+        for (let i = 0; i < headerList.length; i++) {
```
Чтобы проверить есть ли в inbox письмо с заданным title, не нужен цикл.
Нужно выстроить xpath локатор, который будет искать элемент по тексту.
Если на странице есть header элемент с таким текстом, то проверка пройдена.

kateyurasova commented on Pages/inboxPage.js:
```diff
@@ -0,0 +1,68 @@
+        for (let i = 0; i < headerList.length; i++) {
+            if (headerList[i] == RandomTitle) {
+                //expect(headerList.get(i).getText()).toBe(RandomTitle);
+                expect(list.get(i).getText()).toBe(' - ' + BodyText + "EXTRA INFO FOR TEST");
```
Проверки body должна также происходить по наличию определенного локатора. Выстраивать локатор важно с учетом header элемента с определенным текстом.

kateyurasova commented on Pages/inboxPage.js:
```diff
@@ -0,0 +1,68 @@
+    let Title = 0;
+    this.typeRandomTitle = function () {
+        for (let i = 0; i < 10; i++) {
+            Title += chance.letter()
```
у библитеки chance уже есть готовый метод для генерации строки. Используй его. Цикл здесь не нужен.
https://chancejs.com/basics/string.html

kateyurasova commented on Pages/inboxPage.js:
```diff
@@ -0,0 +1,68 @@
+    };
+
+    let Title = 0;
+    this.typeRandomTitle = function () {
```
название метода не соответствует его содержимому

kateyurasova commented on Tests/spec.js:
```diff
@@ -0,0 +1,38 @@
+let data = require("../Data/data.js");
+let loginPage = require("../Pages/loginPage.js");
+let inboxPage = require("../Pages/inboxPage.js");
+let RandomTitle = inboxPage.typeRandomTitle();//генерируем уникальное имя для письма и вводим шаблонный текст письма
```
метод для генерации random title не относится к классу страницы. 
Нужно создать отдельный класс для него если хочешь скрыть детали алгоритма генерации.

kateyurasova commented on Tests/spec.js:
```diff
@@ -0,0 +1,38 @@
+//тесты
+describe('This is first protractor test which', function () {
+    beforeEach(function () {
+        loginPage.open(data.site)
```
Параметр data.site здесь избыточен. Вычитывать значение site нужно в методе open.

kateyurasova commented on Tests/spec.js:
```diff
@@ -0,0 +1,38 @@
+
+    it("should send email and verify Sent email", function () {
+        loginPage.login(data.senderEmail, data.PassOne);//заходим в почту, создаем письмо и отправляем 2му пользователю
+        inboxPage.newDraft();
```
Имя метода должно начитаться с глагола, который однозначно характеризует действие.

kateyurasova commented on Tests/spec.js:
```diff
@@ -0,0 +1,38 @@
+    it("should send email and verify Sent email", function () {
+        loginPage.login(data.senderEmail, data.PassOne);//заходим в почту, создаем письмо и отправляем 2му пользователю
+        inboxPage.newDraft();
+        inboxPage.SendTo(data.receiverEmail);
```
Название метода не отражает суть происходящего.

kateyurasova commented on Tests/spec.js:
```diff
@@ -0,0 +1,38 @@
+        loginPage.login(data.senderEmail, data.PassOne);//заходим в почту, создаем письмо и отправляем 2му пользователю
+        inboxPage.newDraft();
+        inboxPage.SendTo(data.receiverEmail);
+        inboxPage.TypeTitle(RandomTitle);//get data from inboxPage
```
имя метода должно начитаться с маленькой буквы

kateyurasova commented on package.json:
```diff
@@ -15,5 +17,12 @@
+    "chance": "^1.0.16",
+    "decache": "^4.4.1",
+    "jasmine-allure-reporter": "^1.0.2"
+  }
```
Уверена, что в конфиге должен быть протрактор. Добавь, пожалуйста.

kateyurasova commented on Pages/loginPage.js:
```diff
@@ -0,0 +1,35 @@
+    this.login = function (login, password) {
+        this.enterName(login);
+        this.submit();
+        browser.wait(EC.visibilityOf($('div[id="password"] input[name="password"]')), 10000);
```
Вынести локатор в переменную. В таком виде проблематично код плохо читаем

RapeikaD committed:
v0.8.1

+Readme update
+allure reporter
+integration with jenkins

RapeikaD committed:
Remove allure-report folder

katsiaryna-yurasova-zefr commented on package.json:
```diff
@@ -4,7 +4,9 @@
+    "pretest": "rm -rf allure-report",
+    "test": "protractor conf.js",
+    "posttest": "allure generate allure-results --clean -o allure-report || true"
   },
```
"protractor": "protractor conf.js "

RapeikaD committed:
Script update

RapeikaD committed:
Script update v2

RapeikaD committed:
Update package.json

RapeikaD committed:
Update package.json

RapeikaD committed:
Update package.json

RapeikaD committed:
v0.8.3

Small fixes. Preparation for locators updates.

RapeikaD committed:
v0.9

+New Comments in code.
+Logging added .
+Rework of "Email verification" functions.
+Remove excess code.
+"Title" parameter is not hardcoded now.

RapeikaD committed:
v1.0

Rework Title verification function.
Rework Body Text verification function.
Rework Sender Name verification function.
Rename variables.
Add New comments.

RapeikaD committed:
v1.1

Main functions updated.
Dependency of test cases has removed.
Logger added.
Rename some variables.
Excess code removed.

katsiaryna-yurasova-zefr commented on Data/data.js:
```diff
@@ -0,0 +1,11 @@
+"use strict"; //*******
```
Имена папок должны быть с маленькой буквы
